common/tests/unittests/test_hash_pandas_df.py

```diff
@@ -19,9 +19,10 @@ def _get_pandas_df(num_rows=20):
 
 def test_when_df_saved_and_loaded_from_disk_then_hash_is_unchanged():
     df = _get_pandas_df()
-    with TemporaryFile() as temp_file:
-        save_pkl.save(str(temp_file), df)
-        loaded_df = load_pkl.load(str(temp_file))
+    with tempfile.TemporaryDirectory() as _:
+        tem_pfile = "temp_file.pkl"
+        save_pkl.save(tem_pfile, df)
+        loaded_df = load_pkl.load(tem_pfile)
```

Comment:
It seems that the current code saves `temp_file.pkl` outside the `TemporaryDirectory`, is this the case?

Also, probably typo in the file name `tem_pfile` -> `temp_file`?

Would the following code work? (also requires adding `from pathlib import Path` in the header)
```suggestion
    with tempfile.TemporaryDirectory() as temp_dir:
        temp_file = str(Path(temp_dir.name) / "temp_file.pkl")
        save_pkl.save(temp_file, df)
        loaded_df = load_pkl.load(temp_file)
```

Reply:
Nice suggestion. Your code doesn't work out of the box. Made some changes